### PR TITLE
adding yui-ibc-solidity sample

### PR DESCRIPTION
**Pull request by 3100**
As a sample of application development using yui-ibc-solidity,
we created an environment for inter-ledger transfer using a token called MiniToken.

### CODE REVIEW

File: contracts/minitoken/solidity/contracts/app/MiniToken.sol [hunk 1]
```diff
@@ -0,0 +1,272 @@
+pragma experimental ABIEncoderV2;
```
[the remark this answers is not in the document]
Comment (3100) [reply contracts/minitoken/solidity/contracts/app/MiniToken.sol]: @bluele
Thank you, I fixed it.

File: contracts/minitoken/solidity/contracts/app/MiniToken.sol [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+
+    function transfer(address to, uint256 amount) external {
+        bool res;
+        string memory message;
+        (res, message) = _transfer(msg.sender, to, amount);
+        require(res, message);
+    }
+
+    function balanceOf(address account) public view returns (uint256) {
+        require(account != address(0), "Token: invalid address");
```
Comment (siburu) [inline contracts/minitoken/solidity/contracts/app/MiniToken.sol:100]: I know it's a small detail, but why is the case `account == 0` handled specially?
Users can still mint or transfer a token to any non-zero invalid addresses, can't they?

Comment (3100) [reply contracts/minitoken/solidity/contracts/app/MiniToken.sol:100]: @siburu 
You're absolutely right. There's no need for special handling here. Thank you.

**Comment by bluele** [comment]
Can you reassign the review or mention me after all fixes are completed?

File: contracts/minitoken/solidity/contracts/app/MiniToken.sol (hunk 2 above)
Comment (siburu) [reply contracts/minitoken/solidity/contracts/app/MiniToken.sol]: @3100 Thank you and I'm sorry that I have to say that `_mint`, `_burn` and `_transfer` are also checking if `account == 0`. Wouldn't it be better to remove these as well?

Comment (3100) [reply contracts/minitoken/solidity/contracts/app/MiniToken.sol]: @siburu 
Sorry for the delay in responding. Considering that we can apply the MiniToken design to other ledgers, it was strange to consider only the zero address invalid.
Fixed it.

File: contracts/minitoken/solidity/contracts/app/MiniToken.sol [hunk 3]
```diff
@@ -0,0 +1,262 @@
+pragma solidity ^0.8.9;
```
Comment (bluele) [inline contracts/minitoken/solidity/contracts/app/MiniToken.sol:1]: @3100 can we add SPDX license "Apache-2.0" ?

File: contracts/minitoken/solidity/contracts/app/MiniToken.sol [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+        string calldata channelId,
+        string calldata counterpartyVersion
+    ) external virtual override {}
+
+    function onChanOpenConfirm(
+        string calldata portId,
+        string calldata channelId
+    ) external virtual override {}
+
+    function _sendPacket(
```
[the remark this answers is not in the document]
Comment (3100) [reply contracts/minitoken/solidity/contracts/app/MiniToken.sol:200]: Added.

File: contracts/minitoken/solidity/proto/lib/Packet.proto [hunk 5]
```diff
@@ -2,6 +2,9 @@ syntax = "proto3";
 
 option go_package = "pkg/ibc/app";
 
+import "solidity-protobuf-extensions.proto";
+option (.solidity.file_options) = { dirpath: "contracts/lib" };
```
Comment (siburu) [inline contracts/minitoken/solidity/proto/lib/Packet.proto]: @3100 Unfortunately, the `dirpath` option will be discontinued.
https://github.com/datachainlab/solidity-protobuf/issues/32

You can run this example without using `dirpath`, by specifying `--sol_out=...:$(pwd)/contracts/lib`.

Comment (3100) [reply contracts/minitoken/solidity/proto/lib/Packet.proto]: @siburu Thank you. Fixed it as you suggested.

**Review by siburu — APPROVED** [review]
LGTM

**Review by bluele — APPROVED** [review]
LGTM :+1:

**Pull request merged**